### DISCUSSION TITLE
Discussion for a feature to fill an area between line series

### DISCUSSION THREAD

**Discussion by timocov** [tradingview/lightweight-charts#714] [Ideas]
Feature request: https://github.com/tradingview/lightweight-charts/issues/140.

@mastermobin let's continue here.

@tradingview/lightweight-charts please join this discussion as well.

I think, that instead of adding the new type of series we can add something like "fill between series" feature, where you need to provide 2 line series and some options for colors etc.

In this case this "filling" might be optional, you can enable and disable it on the fly without modifying the original series so they can still continue to draw their lines.

What do you think guys?

**Comment by bombjackm** [comment]
I think It's a good idea. We should be able to have 2 coloring options. One if line A is above B and another when B above A.
It will allow to cover many indicators (ichimoku cloud, filled bollinger line, crossed MA...)

**Comment by carloswm85** [comment]
I strongly agree, because I can make good use of that. Lol.

**Comment by georgehenderson** [comment]
It might be nice to support multiple linked series pairs. Imagine an upper and lower Bollinger area and a breakout area between a Keltner upper or lower line and the respective bollinger.
![9F45F775-9989-4953-9903-461E539C86CE](https://user-images.githubusercontent.com/1379792/120228156-2d49ab00-c210-11eb-9f8c-a8dda5324176.jpeg)

**Comment by prirefi** [comment]
any progress on this?

**Comment by aetikon** [comment]
Another option would be to use the areaSeries but require the high and low prices (instead of high and zero), and then fill it in.

**Reply by 0x0tyy** [reply]
is this a viable solution currently? or do you have to implement this yourself?

**Comment by hiCozyty** [comment]
I like this idea as well

**Comment by brahmanzadeh** [comment]
Any progress? I really need a tool to create a shape like ichi cloud.

**Comment by mastermobin** [comment]
I implemented it myself with a few tweaks on the original source back then. I'm not sure if I still have it or not, but it wasn't up to the standards of this repository, so I couldn't merge mine. 
If I found it, I can make it available on my fork. It's not based on the last version, but you can merge it yourself locally.

**Reply by mastermobin** [reply]
It was already there in my fork (look for cloud-area-chart branch).
https://github.com/mastermobin/lightweight-charts.

It would be great if anyone could use this source to implement something more standard, and then submit a pull request, but I don't really have the time right now.

I hope this helps.

**Reply by 0x0tyy** [reply]
thanks for the inspo. was able to come up with my own solution.

https://user-images.githubusercontent.com/108040259/211204193-a9e59324-c215-4414-b822-37d6ce69173c.mp4

**Comment by Patricks15** [comment]
Any news on that feature? For me this is the missing feature to use this library

**Reply by mpasa** [reply]
@Patricks15 I think this should be possible with the new Plugin system introduced in 4.1. There is an example of a band indicator [here](https://tradingview.github.io/lightweight-charts/plugin-examples/plugins/bands-indicator/example/).

**Reply by nitek29** [reply]
@mpasa Band indicator takes only one Series right ? Ichimoku or Bollinger Band needs 2 or 3 Series (upper, middle, lower)
